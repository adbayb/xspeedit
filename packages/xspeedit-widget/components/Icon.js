import React from "react";
import PropTypes from "prop-types";

const Icon = ({ name, width, height, className }) => {
	return (
		<img
			className={className}
			src={`/static/${name}.svg`}
			alt={`icon-${name}`}
			width={width}
			height={height}
		/>
	);
};

Icon.propTypes = {
	name: PropTypes.oneOf(["github", "info", "main", "twitter", "enter"]).isRequired,
	width: PropTypes.number,
	height: PropTypes.number,
	className: PropTypes.string
};

Icon.defaultProps = {
	width: 24,
	height: 24
};

export default Icon;
