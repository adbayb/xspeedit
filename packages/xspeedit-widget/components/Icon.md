React component example:

```js
<div>
	<Icon name="enter" />
	<Icon name="info" />
	<Icon name="github" />
	<Icon name="twitter" />
</div>;
```

You can add a custom props to an example wrapper:

```js { "props": { "className": "checks" } }
<Icon name="info" />;
```

Or disable an editor by passing a `noeditor` modifier:

```jsx noeditor
<Icon name="main" />;
```

To render an example as highlighted source code add a `static` modifier:

```jsx static
import React from 'react';
```
