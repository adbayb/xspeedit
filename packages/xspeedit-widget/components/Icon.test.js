import React from "react";
import Icon from "./Icon.js";

describe("Icon", () => {
	it("should render correctly", () => {
		// GIVEN: notre composant:
		const tree = shallow(<Icon name="enter" />);
		// THEN: il doit s'afficher correctement en matchant son snapshot:
		expect(tree).toMatchSnapshot();
	});
});
