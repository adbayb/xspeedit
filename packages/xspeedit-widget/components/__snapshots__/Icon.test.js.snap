// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<img
  alt="icon-enter"
  height={24}
  src="/static/enter.svg"
  width={24}
/>
`;
